content/courses/advanced-ada/parts/control_flow/exceptions.rst

```diff
@@ -3,19 +3,209 @@ Exceptions
 
 .. include:: ../../../global.txt
 
-..
-    TO BE DONE:
 
-    Classification of Errors
-    ------------------------
+.. _Adv_Ada_Classification_Of_Errors:
 
-    .. admonition:: In the Ada Reference Manual
+Classification of Errors
+------------------------
 
-        - :arm:`1.1.5 Classification of Errors <1-1-5>`
+When we talk about errors and erroneous behavior in Ada, we can classify them
+in one of the four categories:
 
-    .. todo::
+- compilation errors |mdash| i.e. errors that an Ada compiler must detect at
+  compilation time;
+
+- runtime errors |mdash| i.e. errors that are detected by an Ada-based
+  application using checks at runtime;
+
+- bounded errors;
+
+- erroneous execution.
+
+In this section, we discuss each of these categories.
+
+.. admonition:: In the Ada Reference Manual
+
+    - :arm:`1.1.5 Classification of Errors <1-1-5>`
+
+
+.. _Adv_Ada_Compilation_Errors:
+
+Compilation errors
+~~~~~~~~~~~~~~~~~~
+
+In the category of compilation errors, the goal is to prevent compilers to
+accept illegal programs. Here, any program that doesn't follow the rules
```

Comment:
The phrase "the goal is to prevent compilers to accept illegal programs." is a little awkward in English.  Consider as alternatives:
"In the category of compilation errors, the goal is to prevent compilers from accepting illegal programs."
or
"In the category of compilation errors, the goal is to ensure the compiler does not accept illegal programs."

Reply:
Yes, "prevent ... from accepting" was what I actually wanted to write. :-/

Reply:
Thanks for the suggestions!